### PR TITLE
Feature abereziny refactor for cluster usage

### PR DESCRIPTION
**Pull request by abereziny**
- Added Cluster object with hierarcy for nodes in a cluster under test.
- Removed hardcoded env variables, now endpoints, passwords and paths are obtained from hosting
- Added base class with all mandatory dependencies
- Fixed some code issues along the way
- #416 

Signed-off-by: Andrey Berezin <a.berezin@yadro.com>

### CODE REVIEW

File: pytest_tests/helpers/cluster.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    def label(self) -> str:
+        return f"{self.name}: {self.get_endpoint()}"
+
+
+class HTTPGate(NodeBase):
+    """
+    Class represents HTTP gateway in a cluster
+    """
+
+    def get_endpoint(self) -> str:
```
Comment (dansingjulia) [inline pytest_tests/helpers/cluster.py:120]: we can have a set of endpoins for gate.

File: pytest_tests/helpers/cluster.py [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+    pass
+
+
+class S3Gate(NodeBase):
+    """
+    Class represents S3 gateway in a cluster
+    """
+
+    def get_endpoint(self) -> str:
+        return self._get_attribute(_ConfigAttributes.ENDPOINT)
```
Comment (dansingjulia) [inline pytest_tests/helpers/cluster.py:108]: we can have a set of endpoins for gate.

**Review by dansingjulia — APPROVED** [review]

File: pytest_tests/helpers/cluster.py (hunk 1 above)
Comment (abereziny) [reply pytest_tests/helpers/cluster.py:120]: at first we need update testsetup to have multiple addresses in hosting

File: pytest_tests/helpers/cluster.py (hunk 2 above)
Comment (abereziny) [reply pytest_tests/helpers/cluster.py:108]: at first we need update testsetup to have multiple addresses in hosting

**Pull request merged**